File: api/skill/v1/skill.go
// Package v1 Skill API：List、Execute（SSE 流式）
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

// SkillListReq GET /skill/v1/list 获取已启用技能列表
type SkillListReq struct {
	g.Meta `path:"/skill/v1/list" method:"get" summary:"Skill 列表"`
}

type ParamInfo struct {
	Name        string `json:"name"`
	Type        string `json:"type"`
	Description string `json:"description"`
	Required    bool   `json:"required"`
}

type SkillInfo struct {
	ID          string      `json:"id"`
	Name        string      `json:"name"`
	Description string      `json:"description"`
	Category    string      `json:"category"`
	Params      []ParamInfo `json:"params"`
}

type SkillListRes struct {
	Skills []SkillInfo `json:"skills"`
}

// SkillExecuteReq POST /skill/v1/execute 执行技能，SSE 流式返回 step/result
type SkillExecuteReq struct {
	g.Meta  `path:"/skill/v1/execute" method:"post" summary:"执行 Skill"`
	SkillID string         `json:"skill_id" v:"required"`
	Params  map[string]any `json:"params"`
}

type SkillExecuteRes struct{}

File: web/src/services/skill.ts
import api from './api'
import { streamFetch } from '@/utils/sse'
import { ApiResponse } from '@/types'

export interface Skill {
  id: string
  name: string
  description: string
  category: string
  params: { name: string; type: string; description: string; required: boolean }[]
}

export const skillService = {
  list: async () => {
    const res = await api.get<ApiResponse<{ skills: Skill[] }>>('/skill/v1/list')
    return res.data.data.skills
  },

  execute: (
    skillId: string,
    params: Record<string, unknown>,
    onMessage: (type: string, content: string) => void,
    onDone: () => void
  ) => {
    streamFetch(
      '/api/skill/v1/execute',
      { skill_id: skillId, params },
      onMessage,
      onDone
    )
  },
}

